locmap/lib/locMapRESTAPI2.js

```diff
@@ -53,6 +53,99 @@ var LocMapRESTAPI2 = function() {
 
         callback(200, places);
     };
+    this.placeNameAlreadyInUse = function(user, placeName) {
+        for (var key in user.data.places) {
+            if (user.data.places[key].name === placeName) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    this.getVerifiedPlace = function(userId, cache, place, callback) {
+        if (!place || (typeof place !== 'object') || place.lat === undefined || place.lon === undefined ||
+            place.acc === undefined || place.name === undefined || place.img === undefined) {
+            callback(400, 'Place object is wrong!');
+            return;
+        }
+        var verifiedPlace = locMapCommon.verifyPlace(place);
+        if (verifiedPlace) {
+            var user = cache.get('locmapuser', userId);
+            if (Object.keys(user.data.places).length >= conf.get('locMapConfig').maxPlacesLimitNormalUser) {
+                callback(403, 'place_limit_reached');
+                return;
+            }
+
+            if (verifiedPlace.name.length > conf.get('locMapConfig').maxPlaceNameLength) {
+                callback(403, 'place_name_too_long');
+                return;
+            }
+
+            if (this.placeNameAlreadyInUse(user, verifiedPlace.name)) {
+                callback(403, 'place_name_already_in_use');
+                return;
+            }
+        } else {
+            callback(400, 'Couldn\'t verify place!');
+        }
+        return verifiedPlace;
+    }
```

Comment:
This is copy pasta, but since we are going to merge api V2 to V1 at the end of the project, I can live with this.